pkg/Microsoft.NETCore.Platforms/runtimeGroups.props

```diff
@@ -140,14 +140,43 @@
     </RuntimeGroupWithQualifiers>
   </ItemGroup>
 
-  <Target Name="GenerateRuntimeJson" BeforeTargets="CreatePackage">
-    <!-- Note that we intentionally modify the runtime*json that is under source control.
-         We want to keep this up to date in source control so that we can see the diff in the 
-         runtime.json over time as changes are made to RuntimeGroups. -->
-    <MakeDir Directories="$(PackageReportDir)" />
+  <Target Name="GenerateRuntimeJson">
+    <MakeDir Directories="$(IntermediateOutputPath);$(PackageReportDir)" />
     <GenerateRuntimeGraph RuntimeGroups="@(RuntimeGroupWithQualifiers)"
-                          RuntimeJson="runtime.json"
-                          CompatibilityMap="runtime.compatibility.json"
+                          RuntimeJson="$(IntermediateOutputPath)runtime.json"
+                          CompatibilityMap="$(IntermediateOutputPath)runtime.compatibility.json"
                           RuntimeDirectedGraph="$(PackageReportDir)$(Id)$(NuspecSuffix)-runtime.json.dgml" />
+     <ItemGroup>
+       <GeneratedRuntimeFile Include="$(IntermediateOutputPath)runtime.json;$(IntermediateOutputPath)runtime.compatibility.json"/>
+       <GeneratedRuntimeFile>
+         <TargetFile>$(MSBuildProjectDirectory)\%(FileName)%(Extension)</TargetFile>
+       </GeneratedRuntimeFile>
+     </ItemGroup>
+  </Target>
+
+  <!-- CheckRuntimeJson checks if the generated file is in sync with the files checked into source control
+       It runs during all builds and will force a build break if the generated files are not updated -->
+  <Target Name="CheckRuntimeJson"
+          BeforeTargets="CreatePackage"
+          DependsOnTargets="GenerateRuntimeJson">
+    <ItemGroup>
+       <GeneratedRuntimeFile>
+         <UpdateReason Condition="!Exists('%(TargetFile)')">'%(TargetFile)' is missing.</UpdateReason>
+         <UpdateReason Condition="!Exists('%(FullPath)')">'%(FullPath)' is missing: error in GenerateRuntimeGraph.</UpdateReason>
+       </GeneratedRuntimeFile>
+       <GeneratedRuntimeFile Condition="'%(GeneratedRuntimeFile.UpdateReason)' == ''">
+         <UpdateReason Condition="'$([System.IO.File]::ReadAllText(`%(FullPath)`))' != '$([System.IO.File]::ReadAllText(`%(TargetFile)`))'">'%(TargetFile)' is out of date.</UpdateReason>
```

Comment:
This is ugly but the best I could do without writing a custom task.  I made sure to do it all in condition so that the contents don't get spewed to the log.  I also measured it to make sure it's not too expensive (less than 1% of total project build time).

Reply:
This is ugly. I think it would be better to do this in the task.

Reply:
Should we block this PR (which is fixing CI breaks) on that?  I really wanted to get something in ASAP without delaying by plumbing additional stuff through buildtools.

Alternatively I can just turn off the auto-generation target now, and then do a better fix on the buildtools side and let that make its way over.

Reply:
Don't block the fix but I do think it is worth temporarily turning off the auto-update until the buildtools change can flow.

Reply:
This fix does turn off auto-update.  I'm asking if you think I should make a smaller tactical fix (just turn off auto-update) and implement this solution in BuildTools instead.

Reply:
Yes please.